variables.tf

```diff
@@ -87,67 +119,80 @@ variable "worker" {
 }
 
 variable "rhel_image_name" {
+    type        = string
     description = "Name of the RHEL image that you want to use for the bastion node"
     default = "rhel-8.2"
 }
 
 variable "rhcos_image_name" {
+    type        = string
     description = "Name of the RHCOS image that you want to use for OCP nodes"
     default = "rhcos-4.6"
 }
 
 variable "processor_type" {
+    type        = string
     description = "The type of processor mode (shared/dedicated)"
     default = "shared"
 }
 
 variable "system_type" {
+    type        = string
     description = "The type of system (s922/e980)"
     default = "s922"
 }
 
 variable "network_name" {
+    type        = string
     description = "The name of the network to be used for deploy operations"
     default = "my_network_name"
 }
 
 variable "rhel_username" {
+    type    = string
     default = "root"
 }
 
 variable "public_key_file" {
+    type        = string
     description = "Path to public key file"
     # if empty, will default to ${path.cwd}/data/id_rsa.pub
     default     = "~/.ssh/id_rsa.pub"
 }
 
 variable "private_key_file" {
+    type        = string
     description = "Path to private key file"
     # if empty, will default to ${path.cwd}/data/id_rsa
     default     = "~/.ssh/id_rsa"
 }
 
 variable "private_key" {
+    type        = string
     description = "content of private ssh key"
     # if empty string will read contents of file at var.private_key_file
     default = ""
 }
 
 variable "public_key" {
+    type        = string
     description = "Public key"
     # if empty string will read contents of file at var.public_key_file
     default     = ""
 }
 
 variable "rhel_subscription_username" {
+    type    = string
     default = ""
 }
 
 variable "rhel_subscription_password" {
+    type    = string
```

Comment:
Can we add a validation here that `rhel_subscription_password` cannot be empty if `rhel_subscription_username` is defined.

Reply:
oh hold with respect to latest update from @yussufsh